packages/playground/blueprints/src/lib/steps/install-plugin.ts

```diff
@@ -58,110 +59,152 @@ export const installPlugin: StepHandler<InstallPluginStep<File>> = async (
 	{ pluginZipFile, options = {} },
 	progress?
 ) => {
-	progress?.tracker.setCaption(
-		`Installing the ${zipNameToHumanName(pluginZipFile?.name)} plugin`
-	);
+	const zipFileName = pluginZipFile.name.split('/').pop() || 'plugin.zip';
+	const zipNiceName = zipNameToHumanName(zipFileName);
+
+	progress?.tracker.setCaption(`Installing the ${zipNiceName} plugin`);
 	try {
-		const activate = 'activate' in options ? options.activate : true;
+		// Extract to temporary folder so we can find plugin folder name
+
+		const tmpFolder = '/tmp/plugin';
+		const tmpZipPath = `/tmp/${zipFileName}`;
 
-		// Upload it to WordPress
-		const pluginForm = await playground.request({
-			url: '/wp-admin/plugin-install.php?tab=upload',
+		if (await playground.isDir(tmpFolder)) {
+			await playground.unlink(tmpFolder);
```

Comment:
Does that work if there are files inside the `tmpFolder`?

Reply:
Oh you're right, `FS.unlink` does not remove a directory at all. I found that `FS.rmdir` has a `recursive` option since Node.js v13, but [Emscripten's `FS.rmdir`](https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.rmdir) does not - and will throw an error if the directory is not empty. Then I saw that `BasePHP.rmdir` [handles the recursive option](https://github.com/WordPress/wordpress-playground/blob/trunk/packages/php-wasm/universal/src/lib/base-php.ts#L495-L504) so it works on browser side also.

In the new `installAsset` function, it runs `FS.rmdir(tmpFolder, { recursive: true })` at the start (if needed) and end of the install process.